README.md

```diff
@@ -58,50 +67,137 @@ const colors = require(`${themeDirRequire}/tailwindcss/colors`);
-  module.exports = {
+
+module.exports = {
     plugins: [
-     postcssImportHyvaModules,
-     require('postcss-import'),
-     ...other plugins...
- ]
+        postcssImportHyvaModules,
+        require('postcss-import'),
+        ...other PostCSS plugins...
+    ]
 }
 ```
+
 #### Creating a modules' `tailwind-source.css` file
 
 To declare custom CSS, create a file `view/frontend/tailwind/tailwind-source.css` in a module.
-This file will automatically be imported with `@import` at the end of a themes' `tailwind-source.css` file.  
+This file will automatically be imported with `@import` at the end of a themes' `tailwind-source.css` file.
 
-We recommend only using `@import` statements in a module `tailwind-source.css` file, since other modules CSS files might be appended afterwards, and `@import` statements are only allowed before regular CSS declarations. 
+We recommend only using `@import` statements in a module `tailwind-source.css` file,
+since other modules CSS files might be appended afterwards,
+and `@import` statements are only allowed before regular CSS declarations.
 
 
 ### The `hyvaThemesConfig` and `hyvaModuleDirs` exports
 
 These values are intended to be used in custom build tools and in future utilities.
 
+### How to add CSS variables to TailwindCSS using `twVar` and `twProps`
+
+In you Tailwind Config you can opt in to CSS variables with ease by using these functions,
+instead of adding all of the logic needed your self, these functions will handle this for you.
+
+with `twVar` you can add a CSS variable to one or more TailwindCSS tokens, for example like this:
+
+```js
+const { twVar, mergeTailwindConfig } = require('@hyva-themes/hyva-modules');
+
+module.exports = mergeTailwindConfig({
+    theme: {
+        extend: {
+            colors: {
+                primary: {
+                    lighter: twVar('primary-lighter', colors.blue["600"]),
+                    DEFAULT: twVar('primary', colors.blue["700"]),
+                    darker: twVar('primary-darker', colors.blue["800"]),
+                },
+            }
+        }
+    }
+    // The rest of your tailwind config...
+})
+```
+
+this will render any Tailwind color utility class with the following CSS value:
+
+```css
+.bg-primary {
+    --tw-bg-opacity: 1;
+    background-color: color-mix(in srgb, var(--color-primary, #1d4ed8) calc(var(--tw-bg-opacity) * 100%), transparent);
+}
+```
+
+> The method for handling opacity with all color syntaxes is based on the upcoming TailwindCSS v4,
+> here they use the CSS function `color-mix()` to make this possible,
+> we have reused this to make that transition easier in the future.
+
+and with this you can change the value in your CSS or inline in the page with:
+
+```css
+:root {
+    --color-primary: hsl(20 80% 50%);
+}
+```
+
+Now if you don't want to set this for each TailwindCSS token we also offer the functions `twProps`.
+This acts as a wrapper and will use the keys as the name for the CSS variable,
```

Comment:
Please also add a shorter example that clarifies how the variable names are built, maybe with just one or two lines.

Keep the longer example you used below to showcase how it looks in more context.